manager/api/grpc/server.go

```diff
@@ -54,8 +64,64 @@ func (s *grpcServer) Process(stream manager.ManagerService_ProcessServer) error
 			case <-ctx.Done():
 				return nil
 			case req := <-runReqChan:
-				if err := stream.Send(req); err != nil {
-					return err
+				switch msg := req.Message.(type) {
+				case *manager.ServerStreamMessage_RunReq:
+					data, err := proto.Marshal(msg.RunReq)
+					if err != nil {
+						return err
+					}
+					dataBuffer := bytes.NewBuffer(data)
+					buf := make([]byte, bufferSize)
+					for {
+						n, err := dataBuffer.Read(buf)
+						chunk := &manager.ServerStreamMessage{
+							Message: &manager.ServerStreamMessage_RunReqChunks{
+								RunReqChunks: &manager.RunReqChunks{
+									Data: buf[:n],
+								},
+							},
+						}
+
+						if err := stream.Send(chunk); err != nil {
+							return err
+						}
+
+						if err == io.EOF {
+							break
+						}
+					}
+
+				case *manager.ServerStreamMessage_TerminateReq:
+					terminate := &manager.ServerStreamMessage{
+						Message: &manager.ServerStreamMessage_TerminateReq{
+							TerminateReq: msg.TerminateReq,
+						},
+					}
+					if err := stream.Send(terminate); err != nil {
+						return err
+					}
+
+				case *manager.ServerStreamMessage_StopComputation:
+					stopComp := &manager.ServerStreamMessage{
+						Message: &manager.ServerStreamMessage_StopComputation{
+							StopComputation: msg.StopComputation,
+						},
+					}
+					if err := stream.Send(stopComp); err != nil {
+						return err
+					}
+
+				case *manager.ServerStreamMessage_BackendInfoReq:
+					backendInfo := &manager.ServerStreamMessage{
+						Message: &manager.ServerStreamMessage_BackendInfoReq{
+							BackendInfoReq: msg.BackendInfoReq,
+						},
+					}
+					if err := stream.Send(backendInfo); err != nil {
+						return err
+					}
+				default:
+					return ErrUnexpectedMsg
```

Comment:
use the fault case to send all other messages, and no need to create new variables.